File: c_sharp/Models/Organization/LeaveType.cs
using PeerageHRIS.Models.Employee;
using PeerageHRIS.Models.Time;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PeerageHRIS.Models.Organization
{
    public class LeaveType
    {
        [Key]
        [Required]
        [Display(Name = "ID")]
        public string? TypeId { get; set; }

        [Required]
        [Display(Name = "Leave Type")]
        [Column(TypeName = "nvarchar(450)")]
        public string? TypeName { get; set; }
        
        [Required]
        [Display(Name = "Minimum Leave")]
        public int MinimumLeave { get; set; }

        [Required]
        [Display(Name = "Increment")]
        public int Increment { get; set; } = 0;

        [Required]
        [Display(Name = "Incrementing By-Year(s)")]
        public int IncrementingByYear { get; set; } = 0;

        [Required]
        [Display(Name = "Maximum Leave")]
        public int MaximumLeave { get; set; }

        public bool Enabled { get; set; } = true;

        [Display(Name = "Entry By")]
        [Column(TypeName = "nvarchar(450)")]
        public string? EntryBy { get; set; }

        [Display(Name = "Entry Date")]
        public DateTime EntryDate { get; set; }
        public ICollection<LeaveRequest> Leaves { get; set; } = [];
    }
}

File: lib/features/leave_mgmt/domain/leave_type.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'leave_type.freezed.dart';
part 'leave_type.g.dart';

@freezed
abstract class LeaveType with _$LeaveType {
  factory LeaveType({
    required String typeId,
    required String typeName,
    required int minimumLeave,
    required int increment,
    required int incrementingByYear,
    required int maximumLeave,
    required bool enabled,
    required String entryBy,
    required DateTime entryDate,
    required List<dynamic> leaves,
  }) = _LeaveType;

  factory LeaveType.fromJson(Map<String, dynamic> json) =>
      _$LeaveTypeFromJson(json);
}
